source/fab/cli.py

```diff
@@ -72,15 +69,24 @@ def calc_linker_flags():
     return linker, linker_flags
 
 
-def cli_fab():
+def cli_fab(folder: str = '.', kwargs: Optional[Dict] = None):
     """
     Running Fab from the command line will attempt to build the project in the current or given folder.
+    The following params are used for testing. When run normally any parameters will be caught
+    by a common_arg_parser.
+
+    :param folder:
+        source folder (Testing Only)
+    :param kwargs:
+        parameters  ( Testing Only )
 
     """
-    # todo: use common_arg_parser()?
-    arg_parser = ArgumentParser()
-    arg_parser.add_argument('folder', nargs='?', default='.', type=Path)
-    arg_parser.add_argument('--version', action='version', version=f'%(prog)s {fab.__version__}')
-    args = arg_parser.parse_args()
+    kwargs = kwargs or {}
+    if Path(sys.argv[0]).parts[-1] == 'fab':
```

Comment:
This seems to be checking that the executable is called as "fab". Is this correct and under what circumstances would it not be called that?

Reply:
'fab' could , and is called by other things like 'pytest' so therefore the cli arguments may not apply to 'fab' which causes the argparser to fail with an invalid argument message.